### PR TITLE
Label Support for Persisted Codes from @Values

### PR DESCRIPTION
**Pull request by lopezton-anthem**
# Description

Adds support for retrieving the `label` associated with a `ParamValue`'s `code` that has been previously persisted.

Fixes [Discourse Issue #352](http://discourse.oss.antheminc.com/t/display-value-from-values-instead-of-persisted-code-with-gridcolumn-and-fieldvalue/352)

## Overview of Changes

- Added support for `@GridColumn` and `@FieldValue` to display the `label` instead of a `code` from a given `@Values` annotation, when the `code` has been persisted
  - Renamed existing `[nmDisplayValue]` directive logic to `[nmValueStyles]`.
  - Repurposed `[nmDisplayValue]` to be a component
    - `nmDisplayValue` now applies both `@Values` logic and existing `placeholder` logic, otherwise the `leafState` is used
  - Modified `@FieldValue` to use `[nmDisplayValue]`
  - Modified `@GridColumn` to use `[nmDisplayValue]`
- Created `ConfigTestUtils` for Config Annotation test support

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [ ] `DefaultParamValuesHandlerTest`
- [ ] `DefaultParamValuesHandlerUnitTest`

# Additional Notes

Consider the following code:

```java
public static class Foods {
	
	public static class Dog implements Source {
		@Override
		public List<ParamValue> getValues(String paramPath) {
			List<ParamValue> values = new ArrayList<>();
			values.add(new ParamValue("DOG_FOOD_1", "Dog Food 1"));
			return values;
		}
	}
	
	public static class Cat implements Source {
		@Override
		public List<ParamValue> getValues(String paramPath) {
			List<ParamValue> values = new ArrayList<>();
			values.add(new ParamValue("CAT_FOOD_1", "Cat Food 1"));
			return values;
		}
	}
}
```

```java
@Label("Food Name")
@GridColumn(placeholder = "--")
// Foods.All.class contains both sets of values from Dog.class and Cat.class
@Values(Foods.All.class)
@Path
private PetFood petFood;
```

Assume the state of a param for `petFood` is `Food.Cat.CAT_FOOD_1`. Then the value persisted for the core of `petFood` is `CAT_FOOD_1`. The UI will then use the provided `@Values`to determine and display the label `Cat Food 1`, instead of `CAT_FOOD_1`.

See [Discourse Issue #352](http://discourse.oss.antheminc.com/t/display-value-from-values-instead-of-persisted-code-with-gridcolumn-and-fieldvalue/352) for more background on this issue.

### CODE REVIEW

**Comment by openanthem-admin** [comment]
You did it @lopezton-anthem!

Thank you for signing the Anthem Opensource Software CLA.

**Comment by lopezton-anthem** [comment]
@soham-anthem based on our discussion, I removed the `@Values` support for an array and was unable to get the scenario working for `@ValuesConditional` by setting the target path to be self on the "subscribing" parameter.  Still we can utilize the desired functionality by having clients create a combined `Source` implementation for now, that way we don't open it up.

That being said, @soham-anthem there are no server-side changes.

@dinakarmeda if you're good with the UI changes -- this can be merged.

File: nimbus-ui/nimbusui/src/app/components/platform/card/card-details-field.component.html [hunk 1]
```diff
@@ -21,13 +20,12 @@
...
 
         <div class="form-group" *ngIf="!element.config.uiStyles.attributes.inplaceEdit && !isDate(element.config.type.name)">
-           <!-- <label class="" *ngIf="element.config.uiStyles.attributes.showName==true">{{label}}</label> -->
            <nm-input-label *ngIf="element.config.uiStyles.attributes.showName==true"
                 [element]="element" 
                 [required]="false">
             </nm-input-label>
             <div class="textWrapBreakWord" [nmDisplayValue]='element.leafState' [config]='element.config'>
-                <span class="fieldValue">{{value ? value : placeholder}}&nbsp;</span>
+                <span class="fieldValue">{{displayValue}}&nbsp;</span>
```
Comment (sandeepmantha) [inline nimbus-ui/nimbusui/src/app/components/platform/card/card-details-field.component.html]: Would be better if this is transformed using a directive

Comment (lopezton-anthem) [reply nimbus-ui/nimbusui/src/app/components/platform/card/card-details-field.component.html]: @sandeepmantha Fixed in latest commit... will you please review?

**Comment by lopezton-anthem** [comment]
@dinakarmeda this is ready to be merged

File: nimbus-ui/nimbusui/src/app/directives/display-value.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ *  displayValue the can be used in accordance to the template reference.
+ * 
+ */
+@Component({ 
+    selector: '[nmDisplayValue]',
+    template: `
+        <ng-container *ngTemplateOutlet="bodyTemplate; context: { $implicit: displayValue }"></ng-container>
+    `
+})
+export class DisplayValue {
```
Comment (sandeepmantha) [inline nimbus-ui/nimbusui/src/app/directives/display-value.ts]: Do we need to have 2 directives - one for transforming value and the other for styles ?

**Pull request closed without merging**